packages/flutter_tools/lib/src/build_runner/web_fs.dart

```diff
@@ -265,7 +265,16 @@ class AssetServer {
   Directory partFiles;
 
   Future<Response> handle(Request request) async {
-    if (request.url.path.contains('stack_trace_mapper')) {
+    if (request.url.path.endsWith('.html')) {
+      final Uri htmlUri = flutterProject.web.directory.uri.resolveUri(request.url);
```

Comment:
Should we place some basic protection in place for preventing serving files outside the project directory? Unless `resolveUri` already does it, of course.

Reply:
You cannot send a relative uri so this should not be a problem regardless